src/android/java/com/pspdfkit/cordova/CordovaPdfActivity.java

```diff
@@ -14,96 +16,120 @@
+   * activity.
+   */
+  private static CordovaPdfActivity currentActivity;
+  private final CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+  @NonNull
+  private final DocumentListener listener =
+      new SimpleDocumentListener() {
+        @Override
+        public void onDocumentSaved(@NonNull PdfDocument document) {
+          EventDispatcher.getInstance().sendEvent("onDocumentSaved", new JSONObject());
         }
-        pdfFragment.addDocumentListener(listener);
-    }
 
-    private void releaseActivity() {
-        PdfFragment fragment = currentActivity.getPdfFragment();
-        if (fragment != null) {
-            fragment.removeDocumentListener(listener);
+        @Override
+        public void onDocumentSaveFailed(
+            @NonNull PdfDocument document, @NonNull Throwable exception) {
+          try {
+            final JSONObject data = new JSONObject();
+            data.put("message", exception.getMessage());
+            EventDispatcher.getInstance().sendEvent("onDocumentSaveFailed", data);
+          } catch (JSONException e) {
+            e.printStackTrace();
+          }
         }
-        currentActivity = null;
-    }
+      };
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        bindActivity(this);
-    }
+  public static CordovaPdfActivity getCurrentActivity() {
+    return currentActivity;
+  }
 
-    @Override
-    protected void onDestroy() {
-        releaseActivity();
-        super.onDestroy();
+  private void bindActivity(@NonNull final CordovaPdfActivity activity) {
+    checkArgumentNotNull(activity, "activity");
+    if (currentActivity != null) {
+      throw new IllegalStateException(
+          "EventDispatcher only supports a single CordovaPdfActivity at a time.");
     }
-
-    @Override
-    public void finish() {
-        super.finish();
-        // Notify JavaScript listeners that the activity was dismissed.
-        EventDispatcher.getInstance().notifyActivityDismissed();
+    currentActivity = activity;
+    final PdfFragment pdfFragment = currentActivity.getPdfFragment();
+    if (pdfFragment == null) {
+      throw new IllegalStateException(
+          "EventDispatcher only supports binding to activities that have a fragment instance.");
     }
+    pdfFragment.addDocumentListener(listener);
+  }
 
-    /** Dismisses this PDF activity. */
-    public void dismiss() {
-        finish();
+  private void releaseActivity() {
+    PdfFragment fragment = currentActivity.getPdfFragment();
+    if (fragment != null) {
+      fragment.removeDocumentListener(listener);
     }
-
-    public boolean saveDocument() throws IOException {
-        final PdfDocument document = getDocument();
-        if (document != null) {
-            return document.saveIfModified();
-        } else {
-            return false;
-        }
+    currentActivity = null;
+  }
+
+  /**
```

Comment:
Newly added code:
- `addSubscription`
- `disposeSubscriptions`
- and adding `disposeSubscriptions` in `onStop`